### ISSUE TITLE
Converting X3D to VRML

### ISSUE THREAD

**Issue by Symbian9** [castle-engine/view3dscene#37]
In `README.md` there is mention on converting into VRML:

>  You can also convert all models to X3D or VRML.

But if open `*.x3d` file in `view3dscene` there are NO `.wrl` format to choose in *"File > Save as VRML/X3D (classic encoding)..."* — only `*.x3dv` is available to choose.

Also there are NO any CLI-arguments in `tovrmlx3d` to convert input files into VRML format (only X3D):

- https://castle-engine.io/view3dscene.php#section_converting

**Comment by michaliskambi** [comment]
Indeed, that was confusing docs -- we *can* save VRML, but only if you *open* a VRML. That is, we can "pretty print" VRML, we cannot *convert* to VRML from other formats. If you open a VRML 1.0 or 2.0 file ( https://github.com/castle-engine/demo-models contains some `.wrl` files) then view3dscene "Save as VRML/X3D" will offer to save with `.wrl` extension.

I improved various sentences of https://castle-engine.io/view3dscene.php and in README in repo to avoid this confusion.

Note: I do not see the sentence """You can also convert all models to X3D or VRML.""" in view3dscene README.md (in any download from https://castle-engine.io/view3dscene.php or in GitHub repo). It was present on our website though.

**Comment by splace** [comment]
@michaliskambi 

a thought...

can't you, mostly, save as .x3dv, hand change the header/version, remove profile/meta and change the files extension, if it didn't have v3 features. (which is always the case if it had been, backward compatibly, converted from wrl.)

in which case, couldn't this process be automatic? like...

internally 'save' to stream of bytes in classic encoding, with header changes, reload the stream and if error, report unable and why, is successful then save it as valid wrl?

**Comment by michaliskambi** [comment]
> can't you, mostly, save as .x3dv, hand change the header/version, remove profile/meta and change the files extension, if it didn't have v3 features. (which is always the case if it had been, backward compatibly, converted from wrl.)

That is indeed the manual process of "downgrading" from X3D to VRML 2.0. But no, I do not want to automate it or anywhere guarantee that view3dscene/CGE support it, because it is not trivial in the general case (see 2 points below) and also extremely extremely low priority (VRML 97 is more than 20 years old... you should upgrade to X3D by this point, not downgrade back :) ).

- We do not detect if there are "features in this file that work only in X3D, and some nodes are missing from VRML".
- We do not process NURBS nodes to make the downgrade (and we should https://castle-engine.io/x3d_implementation_nurbs.php#section_vrml2_support ).

Bottom line is that we do not implement, and don't plan to implement a conversion to VRML. It was a simple docs mistake that they suggested we did :)
